### PR TITLE
`Functiontypesyntax.arguments` should be named `parameters`

### LINKED ISSUE #1720: `FunctionTypeSyntax.arguments` should be named `parameters`
### Description

The non-generic parameters of a function declaration are *value* parameters, not type parameters. This means the input list of a function type is a parameter list, not an argument list. Contrast this with the *type* arguments of a bound generic type, which are substitutes for the generic type parameters of the corresponding type declaration.

### PR DESCRIPTION
**Pull request by kimdv**
Resolves #1720

### CODE REVIEW

**Comment by kimdv** [comment]
@swift-ci please test

**Review by ahoppen — APPROVED** [review]
Thanks @AnthonyLatsis for reporting this and @kimdv for fixing it.

File: Sources/SwiftSyntax/SwiftSyntaxCompatibility.swift [hunk 1]
```diff
@@ -85,6 +85,60 @@ public extension EnumDeclSyntax {
   }
 }
 
+public extension FunctionTypeSyntax {
+  @available(*, deprecated, renamed: "unexpectedBetweenLeftParenAndParameters")
+  var unexpectedBetweenLeftParenAndArguments: UnexpectedNodesSyntax? { unexpectedBetweenLeftParenAndParameters }
+
+  @available(*, deprecated, renamed: "parameters")
+  var arguments: TupleTypeElementListSyntax { parameters }
```
Comment (ahoppen) [inline Sources/SwiftSyntax/SwiftSyntaxCompatibility.swift:93]: Ah, I totally forgot that these properties also need to be settable. Would you mind also changing all the other properties in this compatibility layer to be settable?

Comment (kimdv) [reply Sources/SwiftSyntax/SwiftSyntaxCompatibility.swift]: Fixed

**Comment by kimdv** [comment]
@swift-ci please test

**Pull request merged**